client/app/scripts/charts/nodes-chart.js

```diff
@@ -391,16 +390,19 @@ class NodesChart extends React.Component {
   }
 }
 
+
 function mapStateToProps(state) {
   return {
-    adjacentNodes: getAdjacentNodes(state),
+    nodes: nodeAdjacenciesSelector(state),
```

Comment:
This is selector is not working as hoped. My test topology has regular node delta `update` events, but no adding/removing. Yet it's going into `updateGraphState` which is a little more expensive.
The trouble is that `nodes-chart:componentWillReceiveProps` is firing a lot, so there should be extra care in not doing too much work there.